R/rebase.R

```diff
@@ -0,0 +1,40 @@
+#' Rebase data handler for straw broom charts with Dygraph
+#'
+#' Draw a straw broom chart.
+#'
+#' @param dygraph Dygraph to draw chart on
+#' @param value Value to rebase to
+#' @param percent If true, ingnore value argument and rebase to percentage difference
+#'
+#' @return Dygraph with specified straw broom chart
+#'
+#' @examples
+#' library(quantmod)
+#' tickers <- c("AAPL", "MSFT")
+#' getSymbols(tickers)
+#' closePrices <- do.call(merge, lapply(tickers, function(x) Cl(get(x))))
+#' dateWindow <- c("2008-01-01", "2009-01-01")
+#' dygraph(closePrices, main = "Value", group = "stock") %>%
+#'   dyRebase(value = 100) %>%
+#'   dyRangeSelector(dateWindow = dateWindow)
+#' dygraph(closePrices, main = "Percent", group = "stock") %>%
+#'   dyRebase(percent = TRUE) %>%
+#'   dyRangeSelector(dateWindow = dateWindow)
+#' dygraph(closePrices, main = "None", group = "stock") %>%
+#'   dyRangeSelector(dateWindow = dateWindow)
+#'
+#' @export
+dyRebase <- function(dygraph, value = 100, percent = FALSE) {
```

Comment:
What's the current naming convention for `dyFoo` functions? Is `Foo` in the name a noun, a name of the chart type? `Rebase` means verb/action describing what the chart is doing over the raw data. We can stick to that convention, in which case `dyStrawBroom` might fit better than `dyRebase`. But please note that `straw broom` is not widely known chart name for it AFAIK. In finance, this type of chart displays returns (rates of changes, e.g. daily) of different financial instruments (e.g. stocks) rebased to common starting index, so in finance it goes by name `relative performance chart` or similar.